addons/wowl/static/src/libs/download.js

```diff
@@ -0,0 +1,191 @@
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
+/**
+ * download.js v4.2, by dandavis; 2008-2018. [MIT] see http://danml.com/download.html for tests/usage
+ * v1 landed a FF+Chrome compat way of downloading strings to local un-named files, upgraded to use a hidden frame and optional mime
+ * v2 added named files via a[download], msSaveBlob, IE (10+) support, and window.URL support for larger+faster saves than dataURLs
+ * v3 added dataURL and Blob Input, bind-toggle arity, and legacy dataURL fallback was improved with force-download mime and base64 support. 3.1 improved safari handling.
+ * v4 adds AMD/UMD, commonJS, and plain browser support
+ * v4.1 adds url download capability via solo URL argument (same domain/CORS only)
+ * v4.2 adds semantic variable names, long (over 2MB) dataURL support, and hidden by default temp anchors
+ *
+ * Slightly modified for export and lint compliance
+ *
+ * @param {Blob | File | String} data
+ * @param {String} [filename]
+ * @param {String} [mimetype]
+ */
+export default function download(data, filename, mimetype) {
+  let self = window, // this script is only for browsers anyway...
+    defaultMime = "application/octet-stream", // this default mime also triggers iframe downloads
+    mimeType = mimetype || defaultMime,
+    payload = data,
+    url = !filename && !mimetype && payload,
+    anchor = document.createElement("a"),
+    toString = function (a) {
+      return String(a);
+    },
+    myBlob = self.Blob || self.MozBlob || self.WebKitBlob || toString,
+    fileName = filename || "download",
+    blob,
+    reader;
+  myBlob = myBlob.call ? myBlob.bind(self) : Blob;
+
+  if (String(this) === "true") {
+    //reverse arguments, allowing download.bind(true, "text/xml", "export.xml") to act as a callback
+    payload = [payload, mimeType];
+    mimeType = payload[0];
+    payload = payload[1];
+  }
+
+  if (url && url.length < 2048) {
+    // if no filename and no mime, assume a url was passed as the only argument
+    fileName = url.split("/").pop().split("?")[0];
+    anchor.href = url; // assign href prop to temp anchor
+    if (anchor.href.indexOf(url) !== -1) {
+      // if the browser determines that it's a potentially valid url path:
+      let ajax = new XMLHttpRequest();
```

Comment:
isn't this the fetch api ?

Reply:
@kebeclibre Careful, this file is an imported library. I just change some syntax stuff, but I would change its logic.